### DISCUSSION TITLE
Tree-sitter understanding

### DISCUSSION THREAD

**Discussion by hello-users9** [zed-industries/zed#21244] [Feature Requests]
Hi. The documentation (https://zed.dev/docs/extensions/languages) does not answer all questions. So, my questions are:
1) If we want to add tree-sitter queries for some language, the [grammars.language]<repo,commit> section is required?
2) From a user experience point of view, I would like the default grammar to be used when adding tree-sitter queries, unless I import a different one. 
3) Why are tree-sitter queries described in Zed (https://github.com/zed-industries/zed/blob/main/crates/languages/src/go/highlights.scm)? The default grammars used in Zed already have queries described. For example, https://github.com/tree-sitter/tree-sitter-go/blob/master/queries/highlights.scm. This is a bit confusing. Which queries have higher priority? 

I'd like to get the answers here so other users can find them later. And it would be nice if this could then be added to the documentation 🙂

**Comment by hello-users9** [comment]
@maxbrunsfeld

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
